### ISSUE TITLE
setLabels doesn't work with donut chart

### ISSUE THREAD

**Issue by rohan-nerdster** [ArielMejiaDev/larapex-charts#83]
The chart still shows the default legend instead of using the given array of labels
<img width="46" alt="image" src="https://github.com/ArielMejiaDev/larapex-charts/assets/145998760/9cffc370-7fa5-4633-b373-5ae551024670">

**Comment by ArielMejiaDev** [comment]
Thanks for adding the issue, I am going to check this as soon as possible.

**Comment by chimit** [comment]
The same with PieChart.

**Comment by lukejames1111** [comment]
How are you populating the labels? It works fine for me using a donut chart:

`->setLabels(['label' one, 'label two']);`

**Comment by chimit** [comment]
Yes, exactly like this:
```php
app(LarapexChart::class)->donutChart()
            ->addData([1, 2])
            ->setLabels(['asd', 'dsa']);
```

<img width="323" alt="image" src="https://github.com/ArielMejiaDev/larapex-charts/assets/839349/bf45e2e1-6610-44b2-9e59-11d91095a95d">

**Comment by lukejames1111** [comment]
Hmm, only difference I have is that I'm calling it slightly differently, but this works for me:

```
    public function build(array $data, array $labels): \ArielMejiaDev\LarapexCharts\DonutChart
    {
        return $this->chart->donutChart()
            ->setTitle('Valuations Outcome')
            ->addData($data)
            ->setLabels($labels);
    }
```

**Comment by chimit** [comment]
No, nothing helps.

```php
public function build(): \ArielMejiaDev\LarapexCharts\DonutChart
{
    return (new LarapexChart)->donutChart()
        ->setTitle('Valuations Outcome')
        ->addData([1, 2, 3, 4])
        ->setLabels(['Label 1', 'Label 2', 'Label 3', 'Label 4']);
}
```

Are you on the latest version of Larapex charts?

**Comment by lukejames1111** [comment]
Yes, installed it today. Did you create the chart using `php artisan make:chart ChartName`? Can you share your app/Charts and your method which you use to call it?

**Comment by chimit** [comment]
I'm using Livewire:

```php
// app/Livewire/MyComponent.php

<?php

namespace App\Livewire;

use ArielMejiaDev\LarapexCharts\DonutChart;
use ArielMejiaDev\LarapexCharts\LarapexChart;
use Livewire\Component;

class MyComponent extends Component
{
    public function build(): DonutChart
    {
        return app(LarapexChart::class)->donutChart()
            ->setTitle('Some title')
            ->addData([1, 2, 3])
            ->setLabels(['Label 1', 'Label 2', 'Label 3']);
    }

    public function render()
    {
        return view('livewire.my-component', ['chart' => $this->build()]);
    }
}
```

```php
// resources/views/livewire/my-component.blade.php

<div>
    {!! $chart->container() !!}

    {{ $chart->script() }}
</div>
```

**Comment by blisstechllc** [comment]
I can confirm I am having the same issue with the labels. I am using livewire as well with a brand new install.

**Comment by HubertAlva** [comment]
Im having the same issue

**Comment by blisstechllc** [comment]
Yes I am having the same issue with pie charts.

**Comment by ArielMejiaDev** [comment]
I would check this issue, sorry for the delay, thanks

**Comment by blisstechllc** [comment]
Sorry I am still not seeing the labels update for pie or donut charts.

`
public function build(): \ArielMejiaDev\LarapexCharts\PieChart
    {
        $reminder_sent = Reminder::where('sent', 'y')->count();
        $reminder_not_sent = Reminder::where('sent', 'n')->count();

        return $this->chart->pieChart()
            ->setTitle('Notifications sent.')
            ->setSubtitle('09/01/23 - '.Date("m/d/Y"))
            ->addData([$reminder_sent, $reminder_not_sent])
            ->setLabels(['Sent', 'Not Sent']);
    }
`

**Comment by blisstechllc** [comment]
I did notice if you remove the conditional on lines 23 and 25 in larapex-charts/stubs/resources/views/chart/script.blade.php the labels start to work.

**Comment by nicosalvadore** [comment]
Hello,
I have the same issue, the labels are not set and the default values are displayed.

**Comment by marineusde** [comment]
I published a fork of the project some days ago and have done some codestyle, bugfixed ect. I tested your code in a laravel 10 project in my version 1.2.3, maybe you can use it too:

https://github.com/marineusde/larapex-charts

working code from @blisstechllc :

```
(new DonutChart())
            ->setTitle('Notifications sent.')
            ->setSubtitle('09/01/23 - '.Date("m/d/Y"))
            ->addData([5, 10])
            ->setLabels(['Sent', 'Not Sent']);
```

![Bildschirmfoto vom 2024-04-24 12-25-40](https://github.com/ArielMejiaDev/larapex-charts/assets/105713989/f41b15bb-f095-4c0e-ad41-9cf1f13b79d3)

**Comment by cdterry87** [comment]
@ArielMejiaDev 

From what I can tell, in `stubs/resources/views/chart/script.blade.php` the following code is what is causing this issue:

```
@if ($chart->labels())
    labels: {!! json_encode($chart->labels(), true) !!},
@endif
```

When it renders in the blade view by calling `$myTable->script()`, it generates the following code for the label:

```
...
<!--[if BLOCK]><![endif]-->            labels: ["Males","Females"],
        <!--[if ENDBLOCK]><![endif]-->
...
```

The simplest fix is just adding an extra line after the initial `@if` like this:

```
@if ($chart->labels())

    labels: {!! json_encode($chart->labels(), true) !!},
@endif
```

Removing the `@if` check also fixes the issue for my tables, but I'm not sure if that causes any other issues if labels aren't present.  Also in this file are the `stacked` and `stroke` properties that may be having similar issues since they also have `@if` checks around them and can probably be resolved the same way I mentioned above.

**Comment by khalidyogah** [comment]
sorry the issue still exist

**Comment by marineusde** [comment]
> sorry the issue still exist

Did you test my fork, it still works there: https://github.com/marineusde/larapex-charts
